### PR TITLE
Cleanup platform dependent code in FontContext

### PR DESCRIPTION
**Pull request by karimnaaji**
- Remove platform dependent code from core library
- Don't load default font resources twice when loading a remote scene

### CODE REVIEW

**Comment by tallytalwar** [comment]
Reviewing this now.

File: android/tangram/src/com/mapzen/tangram/FontFileParser.java [hunk 1]
```diff
@@ -135,7 +135,14 @@ private void processDocument(XmlPullParser parser) throws XmlPullParserException
                                 fallbackFontDict.put(weight, new Vector<String>());
                             }
 
-                            fallbackFontDict.get(weight).add(filename);
+                            // Don't use UI fonts
+                            if (filename.indexOf("UI-") >= 0) {
+                                continue;
+                            }
+
+                            String fullFileName = "system/fonts/" + filename;
```
Comment (tallytalwar) [inline android/tangram/src/com/mapzen/tangram/FontFileParser.java]: We can probably add a constant for `/system/fonts/` and use that constant everywhere in this file.

Comment (karimnaaji) [inline android/tangram/src/com/mapzen/tangram/FontFileParser.java]: Yep, will update that.

**Comment by tallytalwar** [comment]
Ok, this is good once one small comment is addressed.
Tizen code changes should also be simple to handle here. Will rebase tizen branch once this gets merged.

File: core/src/text/fontContext.cpp [hunk 2]
```diff
@@ -14,98 +14,95 @@
...
+
+            auto addFaces = [&](const char* path) {
+                size_t dataSize;
+                char* data = reinterpret_cast<char*>(bytesFromFile(path, dataSize));
+
+                if (data) {
+                    LOG("Adding bundled font at path %s", path);
+
+                    for (int i = 0, size = BASE_SIZE; i < MAX_STEPS; i++, size += STEP_SIZE) {
+                        m_font[i]->addFace(m_alfons.addFontFace(alfons::InputSource(data, dataSize), size));
```
Comment (tallytalwar) [inline core/src/text/fontContext.cpp]: So, with this change if default font could not be added, this line will crash. I hit this issue when getting tizen branch upto date with master.

**Pull request merged**